src/openzaak/components/autorisaties/api/serializers.py

```diff
@@ -37,7 +104,56 @@ def create(self, validated_data):
 
     @transaction.atomic
     def update(self, instance, validated_data):
+        # Because CatalogusAutorisaties cannot be managed via the API, we delete them
+        # to avoid conflicts between regular Autorisaties and CatalogusAutorisaties
+        if (
+            "autorisaties" in validated_data
+            and instance.catalogusautorisatie_set.exists()
+        ):
+            logger.info(
+                "Updating Applicatie %s via the API, deleting existing CatalogusAutorisaties"
+            )
+            instance.catalogusautorisatie_set.all().delete()
+
         applicatie = super().update(instance, validated_data)
         self.create_missing_credentials(applicatie)
 
         return applicatie
+
+    def validate(self, attrs):
```

Comment:
On the second though I now think that maybe it's not needed, because you remove all related `catalogusautorisatie` in the `update` method, so there won't be any overlap, because you will delete them. I think this solution is nice and neat, if you have PUT or PATCH then just recreate Applicatie with individual autorisaties from request body and that's it.

Reply:
Hmm yeah you're right, I'll remove it and when I update the documentation I'll add a warning to say that CatalogusAutorisaties are deleted if you try to modify the Applicatie via the API (created a separate issue for docs https://github.com/open-zaak/open-zaak/issues/1742)